source/common/network/connection_impl.cc

```diff
@@ -39,68 +40,88 @@ void ConnectionImplUtility::updateBufferStats(uint64_t delta, uint64_t new_total
   }
 }
 
-void DynamicSocketOptionsImpl::setSocketRecvBufferSize(uint32_t buffer_size) const {
-  int rc = 0;
-  if (io_handle_.fd() == -1) {
+Api::SysCallIntResult DynamicSocketOptionsImpl::setSocketRecvBufferSize(int buffer_size) const {
+  int rc = -1;
+  int errsrv = -1;
+  socklen_t option_len = sizeof(int);
+  if (-1 == io_handle_.fd()) {
     ENVOY_CONN_LOG(trace, "Stale socket file handle, operation Set Receive Buffer aborted", *this);
-    return;
+    return {rc, errsrv};
   }
   // Modify the receive buffer size to 'buffer_size'
-  rc = setsockopt(io_handle_.fd(), SOL_SOCKET, SO_RCVBUF, &buffer_size, sizeof(uint32_t));
-  if (rc == -1) {
+  auto& os_syscalls = Api::OsSysCallsSingleton::get();
+  Api::SysCallIntResult result =
+      os_syscalls.setsockopt(io_handle_.fd(), SOL_SOCKET, SO_RCVBUF, &buffer_size, option_len);
+  if (-1 == result.rc_) {
     ENVOY_CONN_LOG(trace, "Failed to modify socket receive buffer to size: buffer_size={}", *this,
```

Comment:
Ditto - I like the debug logging - I'd argue it's worth moving this to the os syscalls too so we get the benefit any time we fail to set a socket option